include/stdexec/__detail/__intrusive_ptr.hpp

```diff
@@ -72,32 +102,58 @@ namespace stdexec {
         return *reinterpret_cast<_Ty*>(__value_);
       }
 
-      void __inc_ref_() noexcept {
-        __refcount_.fetch_add(1, std::memory_order_relaxed);
+      __bits_t __inc_ref_() noexcept {
+        auto __old = __ref_count_.fetch_add(__ref_count_increment, std::memory_order_relaxed);
+        return static_cast<__bits_t>(__old);
       }
 
-      void __dec_ref_() noexcept {
-        if (1u == __refcount_.fetch_sub(1, std::memory_order_release)) {
-          std::atomic_thread_fence(std::memory_order_acquire);
-          // TSan does not support std::atomic_thread_fence, so we
-          // need to use the TSan-specific __tsan_acquire instead:
-          STDEXEC_TSAN(__tsan_acquire(&__refcount_));
+      __bits_t __dec_ref_() noexcept {
+        auto __old = __ref_count_.fetch_sub(__ref_count_increment, std::memory_order_acq_rel);
+        if (__count(static_cast<__bits_t>(__old)) == 1) {
           delete this;
```

Comment:
What takes care of destroying the object in `__value_`?

Reply:
pixies. destructive pixies.

Reply:
actually, the `__control_block` destructor runs the destructor for `__value()`.